### PR TITLE
ci: created ockam-artifact.yml

### PR DESCRIPTION
**Pull request by rghdrizzle**
This PR solves the issue #4639 
So I created a workflow to build artifact image of ockam command everytime when there is a change in implementations/rust/ockam path , after building the image , the workflow pushes the image to the container registry.

## Checks

- [x] All commits in this Pull Request are [signed](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) and Verified by Github.
- [x] All commits in this Pull Request follow the Ockam [commit message convention](https://github.com/build-trust/.github/blob/main/CONTRIBUTING.md#commit-messages).
- [x] I accept the Ockam Community [Code of Conduct](https://github.com/build-trust/.github/blob/main/CODE_OF_CONDUCT.md).
- [x] I have accepted the Ockam [Contributor License Agreement](https://github.com/build-trust/ockam-contributors/blob/main/CLA.md) by adding my Git/Github details in a row at the end of the [CONTRIBUTORS.csv](https://github.com/build-trust/ockam-contributors/blob/main/CONTRIBUTORS.csv) file in a separate pull request to the [build-trust/ockam-contributors](https://github.com/build-trust/ockam-contributors) repository. The easiest way to do this is to [edit the CONTRIBUTORS.csv](https://github.com/build-trust/ockam-contributors/edit/main/CONTRIBUTORS.csv) file in the github web UI and create a PR, this will mark the commit as verified.

### CODE REVIEW

File: .github/workflows/ockam-artifact.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
+name: docker-ockam-command-artifact
+
+on:
+  push:
+    branches:
+      - develop
+    paths:
+      - '.implementations/rust/ockam'
```
Comment (metaclips) [inline .github/workflows/ockam-artifact.yml]: This should be `implementations/rust/ockam` let's remove the `.` prefix

Comment (metaclips) [reply .github/workflows/ockam-artifact.yml]: We should also run workflow when there's an update to the dockerfile

File: .github/workflows/ockam-artifact.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+jobs:
+  build_artifact:
```
Comment (metaclips) [inline .github/workflows/ockam-artifact.yml:17]: Let's add `packages: write` and `contents: read` [permission](https://docs.github.com/en/actions/using-jobs/assigning-permissions-to-jobs) to this job

File: .github/workflows/ockam-artifact.yml [hunk 3]
```diff
@@ -0,0 +1,32 @@
+name: docker-ockam-command-artifact
```
Comment (metaclips) [inline .github/workflows/ockam-artifact.yml]: Let's change this to `Ockam Command Docker Artifact Build`

File: .github/workflows/ockam-artifact.yml [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+on:
+  push:
+    branches:
+      - develop
+    paths:
+      - '.implementations/rust/ockam'
+jobs:
+  build_artifact:
+    runs-on: ubuntu-latest
```
Comment (metaclips) [inline .github/workflows/ockam-artifact.yml]: Let's pin the OS to `ubuntu-20.04`

File: .github/workflows/ockam-artifact.yml [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+        password: ${{ secrets.GITHUB_TOKEN }}
+    - id: buildx
+      uses: docker/setup-buildx-action@4b4e9c3e2d4531116a6f8ba8e71fc6e2cb6e6c8c
+      with:
+        driver-opts: |
+          image=moby/buildkit:v0.10.6
+    - name: Build and Push Docker image to container registry
+      run: |
+        docker buildx build --push \
+            --tag ghcr.io/${ORGANIZATION}/${ARTIFACT_NAME}:latest \
```
Comment (metaclips) [inline .github/workflows/ockam-artifact.yml]: `${ARTIFACT_NAME}` and `${ORGANIZATION}` are not defined.

Comment (metaclips) [inline .github/workflows/ockam-artifact.yml]: Can we also tag the image as `ghcr.io/${ORGANIZATION}/${ARTIFACT_NAME}:commit_sha_that_started_build-month-day-year`. This should be `ghcr.io/${ORGANIZATION}/${ARTIFACT_NAME}:${{ github.event.pull_request.head.sha }}-$(date +'%b-%d-%Y')`

**Review by metaclips — CHANGES_REQUESTED** [review]
Thanks for an amazing PR, changes needed for a merge. We also need to rebase this PR, the command below should suffice.
```bash
git pull --rebase https://github.com/build-trust/ockam.git develop
git push -f
```

File: .github/workflows/ockam-artifact.yml (hunk 5 above)
Comment (rghdrizzle) [reply .github/workflows/ockam-artifact.yml]: ah yes , forgot to declare those

**Comment by rghdrizzle** [comment]
> Thanks for an amazing PR, changes needed for a merge. We also need to rebase this PR, the command below should suffice.
> 
> ```shell
> git pull --rebase https://github.com/build-trust/ockam.git develop
> git push -f
> ```

changes have been made .

File: .github/workflows/ockam-artifact.yml [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    branches:
+      - develop
+    paths:
+      - 'implementations/rust/ockam'
+      - 'tools/docker/ockam'
+permissions:
+  contents: read
+  packages: write
+env:
+  ARTIFACT_NAME: ockam-healthcheck
```
Comment (metaclips) [inline .github/workflows/ockam-artifact.yml]: Let's change this to `ockam-artifacts`

**Review by metaclips — CHANGES_REQUESTED** [review]
Little change left

File: .github/workflows/ockam-artifact.yml (hunk 6 above)
Comment (metaclips) [reply .github/workflows/ockam-artifact.yml]: Let's also add a new line at the end of this workflow

Comment (rghdrizzle) [reply .github/workflows/ockam-artifact.yml]: Done

**Comment by rghdrizzle** [comment]
I added a new line , why is still saying there is no new line in the end ?

File: .github/workflows/ockam-artifact.yml [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+        driver-opts: |
+          image=moby/buildkit:v0.10.6
+    - name: Build and Push Docker image to container registry
+      run: |
+        docker buildx build --push \
+            --tag ghcr.io/${ORGANIZATION}/${ARTIFACT_NAME}:${{ github.event.pull_request.head.sha }}-$(date +'%b-%d-%Y') \
+            --file Dockerfile \
+            --platform linux/amd64 .
+
+
```
Comment (metaclips) [inline .github/workflows/ockam-artifact.yml]: There are two new lines and the last line has a tab identation.

**Comment by davide-baldo** [comment]
> I added a new line , why is still saying there is no new line in the end ?

it seems like the last line contains spaces, there shouldn't be any

**Comment by rghdrizzle** [comment]
@davide-baldo Should I squash the commits ? since the workflows shows an error

**Comment by rghdrizzle** [comment]
> @davide-baldo Should I squash the commits ? since the workflows shows an error

Well I squashed it anyways .

**Comment by davide-baldo** [comment]
well done!

**Comment by davide-baldo** [comment]
before merging, I noticed a submodule in the PR, could you tell me if is it necessary?

**Comment by rghdrizzle** [comment]
> before merging, I noticed a submodule in the PR, could you tell me if is it necessary?

Im not sure if its necessary since it automatically added that when I was committing for some reason .

**Comment by rghdrizzle** [comment]
> > before merging, I noticed a submodule in the PR, could you tell me if is it necessary?
> 
> Im not sure if its necessary since it automatically got added when I was committing for some reason .

**Review by metaclips — APPROVED** [review]
Approved, thanks @rghdrizzle

**Pull request merged**